commands/utils/packagehandlers/yarnpackagehandler.go

```diff
@@ -19,5 +33,50 @@ func (yarn *YarnPackageHandler) UpdateDependency(vulnDetails *utils.Vulnerabilit
 }
 
 func (yarn *YarnPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails, extraArgs ...string) (err error) {
-	return yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	executableYarnVersion, err := getYarnVersion()
+	if err != nil {
+		return
+	}
+
+	var tmpDirToDelete string
+	isV2AndAbove := version.NewVersion(executableYarnVersion).Compare(yarnV2Version) <= 0
+
+	if isV2AndAbove {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV2PackageUpdateCmd)
+	} else {
+		vulnDetails.Technology.SetPackageInstallationCommand(yarnV1PackageUpdateCmd)
+		tmpDirToDelete, err = fileutils.CreateTempDir()
+		if err != nil {
+			return
+		}
+		extraArgs = append(extraArgs, modulesFolderFlag+tmpDirToDelete)
+	}
+	err = yarn.CommonPackageHandler.UpdateDependency(vulnDetails, extraArgs...)
+	if err != nil {
+		return
+	}
+
+	if !isV2AndAbove {
+		err = fileutils.RemoveTempDir(tmpDirToDelete)
```

Comment:
why only for v1?

Reply:
because v2 and above doesn't have the problem of node_modules creation so it works fine with the common flow and I don't create a temp dir for it (so no need to delete it)